ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -91,4 +378,411 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
+          "terms": vocabulary.terms
+        }
+      ];
+      var done = assert.async();
+      var updateParent = function() {};
+      var afterMount = function(component) {
+        var formNode = React.addons.TestUtils.
+          findRenderedDOMComponentWithClass(
+            component,
+            'form-horizontal'
+          );
+        assert.ok(formNode);
+        //test form submission
+        var inputNodes = React.addons.TestUtils.
+          scryRenderedDOMComponentsWithTag(
+            formNode,
+            'input'
+          );
+        assert.equal(inputNodes.length, 4);
+        var inputVocabularyName = inputNodes[0];
+        var inputVocabularyDesc =  inputNodes[1];
+        var checkboxCourse =  inputNodes[2];
+
+        React.addons.TestUtils.Simulate.change(
+          inputVocabularyName, {target: {value: "TestB"}}
+        );
+        React.addons.TestUtils.Simulate.change(
+          inputVocabularyDesc, {target: {value: "TestB"}}
+        );
+        React.addons.TestUtils.Simulate.change(
+          checkboxCourse,
+          {target: {value: 'course', checked: true}}
+        );
+        React.addons.TestUtils.Simulate.submit(formNode);
+        waitForAjax(1, function() {
+          assert.equal(//Error is caused by a 400 status code
+            component.state.errorMessage,
+            "There was a problem adding the Vocabulary."
+          );
+          done();
+        });
+      };
+      React.addons.TestUtils.
+        renderIntoDocument(
+          <AddVocabulary
+            vocabularies={vocabularies}
+            repoSlug="repo2"
+            updateParent={updateParent}
+            ref={afterMount}
+          />
+        );
+    }
+  );
+
+  QUnit.test('Assert that add term works in TaxonomyComponent',
+    function(assert) {
+      assert.ok(TaxonomyComponent, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var vocabularyWithoutTerms = {
+        "id": 2,
+        "slug": "difficulty2",
+        "name": "difficulty2",
+        "description": "easy",
+        "vocabulary_type": "f",
+        "required": false,
+        "weight": 2147483647,
+      };
+      var done = assert.async();
+      var afterMount = function(component) {
+        assert.equal(
+          component.state.vocabularies.length,
+          0
+        );
+        assert.equal(
+          component.state.learningResourceTypes.length,
+          0
+        );
+        waitForAjax(2, function() {
+          assert.equal(
+            component.state.vocabularies.length,
+            1
+          );
+          assert.equal(
+            component.state.vocabularies[0].terms.length,
+            2
+          );
+          assertAddTermCommon(assert, component);
+          // Adding second Vocabulary to make sure term added to correct vocabulary
+          component.addVocabulary(vocabularyWithoutTerms);
```

Comment:
Since `addVocabulary` calls `setState` and `setState` is technically asynchronous you will need to put the rest of the code in this function inside `component.forceUpdate(function() {... })`